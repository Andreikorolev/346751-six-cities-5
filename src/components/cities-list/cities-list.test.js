import React from "react";
import renderer from "react-test-renderer";
import CitiesList from "./cities-list";

const noop = () => {};

describe(`CitiesList render correctly`, () => {
  it(`With empty city`, () => {
    const tree = renderer
      .create(
          <CitiesList
            city={``}
            onCityChange={noop}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With Amsterdam city`, () => {
    const tree = renderer
      .create(
          <CitiesList
            city={`Amsterdam`}
            onCityChange={noop}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
