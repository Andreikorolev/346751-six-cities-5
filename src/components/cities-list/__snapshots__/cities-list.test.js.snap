// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList render correctly With Amsterdam city 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Cologne
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Brussels
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Hamburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Dusseldorf
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;

exports[`CitiesList render correctly With empty city 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Cologne
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Brussels
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Hamburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item"
        >
          <span>
            Dusseldorf
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
